### PR TITLE
added endpoints for gettings agency markups and summary amount

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

**Review by hasmikharutyunyan — APPROVED** [review]

File: Api/Models/Agents/BonusSummaryFilter.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+using System;
+
+namespace HappyTravel.Edo.Api.Models.Agents
+{
+    public record BonusSummaryFilter(DateTime? From, DateTime? Till);
```
Comment (kirillta) [inline Api/Models/Agents/BonusSummaryFilter.cs:5]: `Till` isn't clear in that context

File: Api/Services/Markups/DisplayedBonusesService.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+{
+    public class DisplayedBonusesService : IDisplayedBonusesService
+    {
+        public DisplayedBonusesService(EdoContext context)
+        {
+            _context = context;
+        }
+        
+        
+        public IQueryable<Bonus> GetList(AgentContext agentContext)
```
Comment (kirillta) [inline Api/Services/Markups/DisplayedBonusesService.cs:18]: The method name is misleading

File: Api/Models/Agents/BonusSummary.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace HappyTravel.Edo.Api.Models.Agents
+{
+    public readonly struct BonusSummary
+    {
+        public decimal Summary { get; init; }
```
Comment (mingazhev) [inline Api/Models/Agents/BonusSummary.cs:5]: This is money, need to return money amount, with a currency.

File: Api/Models/Agents/BonusSummaryFilter.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Models/Agents/BonusSummaryFilter.cs:5]: Maybe `To`

File: Api/Services/Markups/DisplayedBonusesService.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
+using System.Linq;
+using System.Threading.Tasks;
+using HappyTravel.Edo.Api.Models.Agents;
+using HappyTravel.Edo.Data;
+using Microsoft.EntityFrameworkCore;
+using Prometheus;
+
+namespace HappyTravel.Edo.Api.Services.Markups
+{
+    public class DisplayedBonusesService : IDisplayedBonusesService
```
Comment (mingazhev) [inline Api/Services/Markups/DisplayedBonusesService.cs:10]: MarkupBonusDisplayService maybe?

File: Api/Services/Markups/DisplayedBonusesService.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            return from appliedMarkup in _context.AppliedBookingMarkups
+                join markupPolicy in _context.MarkupPolicies on appliedMarkup.PolicyId equals markupPolicy.Id
+                join agentAgency in _context.AgentAgencyRelations on markupPolicy.AgentId equals agentAgency.AgentId
+                where agentAgency.AgencyId == agentContext.AgencyId
+                select new Bonus
+                {
+                    ReferenceCode = appliedMarkup.ReferenceCode,
+                    Paid = appliedMarkup.Paid,
+                    Amount = appliedMarkup.Amount
```
Comment (mingazhev) [inline Api/Services/Markups/DisplayedBonusesService.cs:28]: lack of currency there

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: Api/Models/Agents/BonusSummaryFilter.cs (hunk 1 above)
Comment (DFSko) [reply Api/Models/Agents/BonusSummaryFilter.cs]: Fixed

File: Api/Services/Markups/DisplayedBonusesService.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Markups/DisplayedBonusesService.cs]: Fixed

File: Api/Controllers/AgentControllers/AgentMarkupsController.cs [hunk 6]
```diff
@@ -120,8 +124,37 @@ public async Task<IActionResult> GetPolicies(int agentId)
         public IActionResult GetPolicyTemplates() => Ok(_policyTemplateService.Get());
 
 
+        /// <summary>
+        ///     Gets list of applied markups for agency
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("agency/bonuses")]
+        [InAgencyPermissions(InAgencyPermissions.ObserveMarkup)]
```
[the remark this answers is not in the document]
Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentMarkupsController.cs:132]: Fixed

File: Api/Models/Agents/BonusSummary.cs (hunk 3 above)
Comment (DFSko) [reply Api/Models/Agents/BonusSummary.cs]: Fixed

File: Api/Services/Markups/DisplayedBonusesService.cs (hunk 4 above)
Comment (DFSko) [reply Api/Services/Markups/DisplayedBonusesService.cs]: Fixed

File: Api/Services/Markups/DisplayedBonusesService.cs (hunk 5 above)
Comment (DFSko) [reply Api/Services/Markups/DisplayedBonusesService.cs]: Fixed

File: Api/Controllers/AgentControllers/AgentMarkupsController.cs [hunk 7]
```diff
@@ -120,8 +124,39 @@ public async Task<IActionResult> GetPolicies(int agentId)
...
+        [ProducesResponseType(typeof(List<Bonus>), (int) HttpStatusCode.OK)]
+        [EnableQuery]
+        public async Task<IActionResult> GetBonuses()
+        {
+            var agent = await _agentContext.GetAgent();
+            return Ok(_markupBonusDisplayService.GetBonusesList(agent));
```
Comment (mingazhev) [inline Api/Controllers/AgentControllers/AgentMarkupsController.cs:138]: Just "GetBonuses" will be enough. This is not a list

**Review by mingazhev — APPROVED** [review]

File: Api/Controllers/AgentControllers/AgentMarkupsController.cs (hunk 7 above)
Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentMarkupsController.cs]: Fixed

**Pull request merged**